### PR TITLE
remove usage of deprecated methods in guava

### PR DESCRIPTION
**Pull request by hzyi-google**

### CODE REVIEW

File: src/main/java/com/google/api/core/ApiFutures.java [hunk 1]
```diff
@@ -77,7 +77,8 @@ public void onSuccess(V v) {
         Futures.catching(
             listenableFutureForApiFuture(input),
             exceptionType,
-            new GaxFunctionToGuavaFunction<X, V>(callback));
+            new GaxFunctionToGuavaFunction<X, V>(callback),
+            directExecutor());
```
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/core/ApiFutures.java]: Add a version of catching that takes an executor, mark this method `@Deprecated`

File: src/main/java/com/google/api/core/ApiFutures.java [hunk 2]
```diff
@@ -95,9 +96,7 @@ public void onSuccess(V v) {
 
   public static <V, X> ApiFuture<X> transform(
       ApiFuture<? extends V> input, final ApiFunction<? super V, ? extends X> function) {
-    return new ListenableFutureToApiFuture<>(
-        Futures.transform(
-            listenableFutureForApiFuture(input), new GaxFunctionToGuavaFunction<V, X>(function)));
+    return transform(input, function, directExecutor());
```
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/core/ApiFutures.java]: Mark this method `@Deprecated`

File: src/main/java/com/google/api/core/ApiFutures.java [hunk 3]
```diff
@@ -126,17 +125,7 @@ public void onSuccess(V v) {
...
-        Futures.transformAsync(
-            listenableInput,
-            new AsyncFunction<I, O>() {
-              @Override
-              public ListenableFuture<O> apply(I input) throws Exception {
-                return listenableFutureForApiFuture(function.apply(input));
-              }
-            });
-    return new ListenableFutureToApiFuture<>(listenableOutput);
+    return transformAsync(input, function, directExecutor());
```
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/core/ApiFutures.java]: Mark this method `@Deprecated`

**Comment by hzyi-google** [comment]
PTAL @garrettjonesgoogle

**Pull request merged**